File: app/src/main/java/com/siva/virtual/data/ImageDao.java
package com.siva.virtual.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.siva.virtual.model.Images;

import java.util.Date;
import java.util.List;

@Dao
public interface ImageDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Images images);

    @Query("UPDATE image_table SET created_date=:date WHERE file_id = :id")
    void updateCreatedTime(Date date, int id);

    @Query("DELETE FROM image_table WHERE file_id = :id")
    void deleteById(int id);

    @Query("SELECT * FROM image_table ORDER BY created_date DESC")
    LiveData<List<Images>> getAllImages();

    @Query("SELECT COUNT() FROM image_table WHERE file_name = :file_name")
    int count(String file_name);

}

File: app/src/main/java/com/siva/virtualWallet/db/ImageDao.kt
package com.siva.virtualWallet.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import java.util.*

@Dao
interface ImageDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(images: Images)

    @Query("UPDATE image_table SET created_date=:date WHERE file_id = :id")
    fun updateCreatedTime(date: Date, id: Int)

    @Query("DELETE FROM image_table WHERE file_id = :id")
    fun deleteById(id: Int)

    @get:Query("SELECT * FROM image_table ORDER BY created_date DESC")
    val allImages: LiveData<List<Images>>

    @Query("SELECT COUNT() FROM image_table WHERE file_name = :file_name")
    fun count(file_name: String): Int
}